tests/test_core_functions.py

```diff
@@ -0,0 +1,69 @@
+import os
+from unittest.mock import patch, call
```

Comment:
Minor point—there's a pytest plugin that makes this syntax slightly more pytest-onic (is that a thing?)

https://github.com/pytest-dev/pytest-mock/

Reply:
It's all syntax sugar, of course. 

Let's you drop the `with` clauses. 

You can even use some [fancy pytest decorators](https://github.com/pytest-dev/pytest-mock/#why-bother-with-a-plugin) to bring the mocked objects out of the unit test function.